src/core/thread/mle_router.cpp

```diff
@@ -2035,6 +2035,7 @@ ThreadError MleRouter::HandleChildUpdateRequest(const Message &aMessage, const I
     }
 
     child->mLastHeard = Timer::GetNow();
+    child->mAddSrcMatchEntryShort = true;
```

Comment:
It seems that device will choose short address when sending data poll once it has one for the moment. Should we leave it as it is for now or do you have any other suggestions?